opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/context/ServiceContextProviderForFintech.java

```diff
@@ -54,6 +58,15 @@
         }
         AuthSession authSession = extractAndValidateAuthSession(request);
         ServiceSession session = extractOrCreateServiceSession(request, authSession);
+
+        if (ProtocolAction.SINGLE_PAYMENT.equals(authSession.getAction().getProtocolAction())
```

Comment:
This should reside inside HBCI protocol. Payment entrypoint is a good place for that as we don't know is it specific to HBCI or Xs2a is affected too